src/olympia/scanners/admin.py

```diff
@@ -390,6 +390,15 @@ class AbstractScannerRuleAdminMixin(admin.ModelAdmin):
     )
     readonly_fields = ('created', 'modified', 'matched_results_link')
 
+    def get_fields(self, request, obj=None):
+        fields = super().get_fields(request, obj)
+        if not self.has_change_permission(request, obj):
+            # Remove the 'definition' field...
+            fields = list(filter(lambda x: x != 'definition', fields))
+            # ...and add its readonly (and pretty!) alter-ego.
+            fields.append('formatted_definition')
+        return fields
```

Comment:
this isn't the _wrong_ way of doing it, though isn't just building the list directly in this function (i.e. not defining `fields =` at the top of the class) more straightforward than a list&filter with a lambda?